### PR TITLE
Update rust 2021, MSRV 1.56.1, `authors = ["Parity Technologies <admin@parity.io>"]`

### PR DESCRIPTION
**Pull request by NukeManDan**
missed one ;)

### CODE REVIEW

File: substrate-archive/src/actors.rs [hunk 1]
```diff
@@ -199,6 +199,7 @@ where
 		self.blocks.send(ReIndex).await?;
 		let actors = self.clone();
 		task::spawn(async move {
+			let _ = &actors;
```
Comment (niklasad1) [inline substrate-archive/src/actors.rs:202]: remove?!

```suggestion
```

Comment (NukeManDan) [reply substrate-archive/src/actors.rs:202]: this is from cargo fix --edition. I would say refactoring or leaving is best, as I think this pattern saves us from a breaking change planned in behavior in this edition

**Comment by NukeManDan** [comment]
MSRV is The "minimum supported Rust version" right?

"increments are a breaking Semver change" policy may applies to this repository 🤔 should it?

> The relevant check should update to the minimal dependency versions first, in order to make sure newer compatible versions are ignored in this regard.

AFAIK minimal stable toolchain version with rust edition 2021 is rustc 1.56.0
https://blog.rust-lang.org/2021/10/21/Rust-1.56.0.html

This could be propagated in all change logs and maybe include a version bump. But what type? patch, minor or major?
Or no bump?

File: substrate-archive/src/actors.rs (hunk 1 above)
Comment (niklasad1) [reply substrate-archive/src/actors.rs:202]: it's not a breaking change rather that the reference is unused thus it's a bug in `cargo fix` FWIW :)

//cc @insipx  ^^

Comment (insipx) [reply substrate-archive/src/actors.rs:202]: Yeah i think removing this is the play. Compiles just fine w/o it

**Review by insipx — APPROVED** [review]

**Comment by insipx** [comment]
> MSRV is The "minimum supported Rust version" right?
> 
> "increments are a breaking Semver change" policy may applies to this repository thinking should it?
> 
> > The relevant check should update to the minimal dependency versions first, in order to make sure newer compatible versions are ignored in this regard.
> 
> AFAIK minimal stable toolchain version with rust edition 2021 is rustc 1.56.0 https://blog.rust-lang.org/2021/10/21/Rust-1.56.0.html
> 
> This could be propagated in all change logs and maybe include a version bump. But what type? patch, minor or major? Or no bump?

We should just do a minor version bump for this, maybe include it in the 0.7 release with the upcoming extrinsic decoding functionality. If you'd like you can add a line to the current CHANGELOG under 'unreleased', otherwise i'll go through and put in all the changes on next release :)

rust 1.56 seems fine to me

**Comment by NukeManDan** [comment]
`rust-version = "1.56.1"` introduced

For reference, I tried to get this to `cargo build` with 1.56.0 and it failed with
```
error: package `...` cannot be built because it requires rustc 1.56.1 or newer, while the currently active rustc version is 1.56.0
```
With `--ignore-rust-version` it _does_ build with 1.56.0, as expected.

File: bin/node-template-archive/Cargo.toml [hunk 2]
```diff
@@ -3,6 +3,7 @@ name = "node-template-archive"
 version = "0.3.0"
 authors = ["Andrew Plaza <aplaza@liquidthink.net>"]
```
Comment (niklasad1) [inline bin/node-template-archive/Cargo.toml:4]: unrelated to this PR: @insipx we should probably as `parity` as authors too to all crates?!

**Review by niklasad1 — APPROVED** [review]

File: bin/node-template-archive/Cargo.toml (hunk 2 above)
Comment (NukeManDan) [reply bin/node-template-archive/Cargo.toml:4]: happy to use whatever, you feel free to commit in this PR's branch too whatever you want. perhaps: 
 https://github.com/paritytech/substrate/blob/master/bin/node/runtime/Cargo.toml#L4
 ```toml
 authors = ["Parity Technologies <admin@parity.io>"]
 ```

Comment (NukeManDan) [reply bin/node-template-archive/Cargo.toml]: updated the last commit to use this throughout, please `git reset HEAD~ --hard` if you think it's not appropriate in this PR (new title)

**Comment by NukeManDan** [comment]
Think this is good to merge? @insipx if you think so, squash it in at your convenience :)

**Review by insipx — APPROVED** [review]

**Pull request merged**